docs/README.md

```diff
@@ -1,21 +1,45 @@
-KubeVirt represents virtual machines as custom resources and manages changes
-to libvirt domains based on the state of those resources.
+Kubernetes allows for extensions to its architecture via
+[*custom resources*]( https://Kubernetes.io/docs/concepts/extend-Kubernetes/api-extension/custom-resources/),
+which add a new endpoint in the Kubernetes API that stores and retrieves a
+collection API objects of a certain kind.
+The *custom resources* by themselves only
+enable store and retrieve structured data;
+to add business logic and specific functionality,
+[*custom controllers*]( https://Kubernetes.io/docs/concepts/extend-Kubernetes/) are needed.
+*Controllers* are clients of the Kubernetes API-Server that typically read an
+object's `.spec`, possibly do things, and then update the object's
+`.status`.
+
+KubeVirt uses CRDs, *controllers* and other Kubernetes features, to
+represent and manage traditional virtual machines side by side with
+containers.
+
+KubeVirt's primary CRD is the VirtualMachine (VM) resource, which contains
+a VirtualMachineInstance (VMI) object. The VMI
+shares similarity with the Pod concept and represents a single virtualized
+workload that executes once until completion (i.e., powered off). In
+addition to the VMI, the key KubeVirt components are the virt-api, the
+virt-controller, the virt-handler, and the virt-launcher.
 
 ### Project Components
 
- * virt-api: This component provides a HTTP RESTful entrypoint to manage
+![KubeVirt components](components.png "Components")
+
+ * **virt-api**: This component provides a HTTP RESTful entrypoint to manage
    the virtual machines within the cluster.
- * virt-controller: This component manages the state of each VMI within the
-   Kubernetes cluster.
- * virt-handler: This is a daemon that runs on each Kubernetes node. It is
+ * **virt-controller**: This component is a Kubernetes Operator that
+ manages the state of each VMI within the Kubernetes cluster. When new VM
+ objects are submitted to the Kubernetes API-Server, this controller takes
+ notice and creates the pod in which the VM will run and delegates the
+ other management operations to the *virt-handler* component.
+ * **virt-handler**: This is a daemon that runs on each Kubernetes node. It is
    responsible for monitoring the state of VMIs according to Kubernetes and
-   ensuring the corresponding libvirt domain is booted or halted accordingly.
- * virt-launcher: This component is a place-holder, one per running VMI. It
-   contains the running VMI and remains running as long as the VMI is defined.
+   ensuring the corresponding libvirt domain is booted or halted accordingly. To perform these operations, the virt-handler signals and send the VM's CRD object to the *virt-launcher* component.
+ * **virt-launcher**: There is one per running VMI. It
+   contains the running VMI and remains running as long as the VMI is defined. It uses a local libvirtd instance within its container to
+   start or terminate the VM.
```

Comment:
virt-launcher doesn't remain running as long as the VMI is defined, it remains running as long as we qemu process is running. The VMI object might still exist in the cluster in a terminated state indefinitely after that point which doesn't indicate a corresponding pod is present.

Perhaps saying "This component directly manages the lifecycle of the qemu process within the VMI's pod and receives lifecycle commands from virt-handler" would be more accurate.

Reply:
fixed :)

Reply:
@davidvossel could you please review the updates?